src/main/java/web/internetshop/dao/jdbcimpl/UserJdbcImpl.java

```diff
@@ -154,4 +151,20 @@ private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
             return roles;
         }
     }
+
+    private void insertUsersRoles(User user, Connection connection) throws SQLException {
+        String selectRoleIdQuery = "SELECT role_id FROM roles WHERE role_name = ?";
+        String insertUsersRolesQuery = "INSERT INTO users_roles (user_id, role_id) VALUES (?, ?);";
+        PreparedStatement selectStatement = connection.prepareStatement(selectRoleIdQuery);
+        for (Role role : user.getRoles()) {
+            selectStatement.setString(1, role.getRoleName().name());
+            ResultSet resultSet = selectStatement.executeQuery();
+            resultSet.next();
+            PreparedStatement insertStatement =
+                    connection.prepareStatement(insertUsersRolesQuery);
```

Comment:
move creating preparedStatement before loop. Look for other similar cases in your project.